source/core/coverage/ut_coverage_proftab.pkb

```diff
@@ -0,0 +1,178 @@
+create or replace package body ut_coverage_proftab is
+  /*
+  utPLSQL - Version 3
+  Copyright 2016 - 2017 utPLSQL Project
+
+  Licensed under the Apache License, Version 2.0 (the "License"):
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+  */
+
+  -- The source query has two important transformations done in it.
+  -- the flag: to_be_skipped ='Y' is set for a line of code that is badly reported by DBMS_PROFILER as executed 0 times.
+  -- This includes lines that are:
+  --   - PACKAGE, PROCEDURE, FUNCTION definition line,
+  --   - BEGIN, END  of a block
+  -- Another transformation is adjustment of line number for TRIGGER body.
+  -- DBMS_PROFILER is reporting line numbers for triggers not as defined in DBA_SOURCE, its usign line numbers as defined in DBA_TRIGGERS
+  -- the DBA_TRIGGERS does not contain the trigger specification lines, only lines that define the trigger body.
+  -- the query adjusts the line numbers for triggers by finding first occurrence of begin|declare|compound in the trigger body line.
+  -- The subquery is optimized by:
+  -- - COALESCE function -> it will execute only for TRIGGERS
+  -- - scalar subquery cache -> it will only execute once for one trigger source code.
+  function get_cov_sources_sql(a_coverage_options ut_coverage_options) return varchar2 is
```

Comment:
Isn't that function duplicated in two packages now?

Reply:
Not exactly, I  block coverage we do not skip any lines. I tried to keep to minimum number of places where there is a If decision. What I might do is to move it to generic helper and in profiler coverage package put some code to apply case statement for skipping lines before execution. So that main query will be maintained in one place.